server/src/main/java/io/crate/metadata/upgrade/IndexTemplateUpgrader.java

```diff
@@ -170,4 +180,92 @@ private static void populateColumnPositions(String parentName,
                                     maxColumnPosition);
         }
     }
+
+
+    /**
+     * Fixes corrupted templates caused by the swap table bug, https://github.com/crate/crate/issues/13380.
+     */
+    @VisibleForTesting
+    static Map<String, IndexTemplateMetadata> fixCorruptedTemplatesFromSwapTableBug(Map<String, IndexTemplateMetadata> templates) {
+        Map<String, IndexTemplateMetadata> fixedTemplates = new HashMap<>();
+        for (Map.Entry<String, IndexTemplateMetadata> e : templates.entrySet()) {
+            // Side effect of the swap table bug:
+            //      ex) alter cluster swap table "m"."t" to "m.s",
+            //          where t and s are partitioned, and two templates named 1) .partitioned.m.t, 2) m..partitioned.t are generated
```

Comment:
Wouldn't the broken template name be `.partitioned.m.s` ?

Reply:
It is weird but the name `s` gets lost. `.partitioned.m.t` is the broken name I see.